python/kserve/kserve/protocol/rest/v2_endpoints.py

```diff
@@ -53,23 +53,27 @@ async def metadata(self) -> ServerMetadataResponse:
         """
         return ServerMetadataResponse.parse_obj(self.dataplane.metadata())
 
-    @staticmethod
-    async def live() -> ServerLiveResponse:
+    async def live(self) -> ServerLiveResponse:
         """Server live endpoint.
 
         Returns:
             ServerLiveResponse: Server live message.
         """
-        return ServerLiveResponse(live=True)
+        response = await self.dataplane.live()
+        # TODO: if server is not live send 5xx error response.
+        is_live = response["status"] == "alive"
+        return ServerLiveResponse(live=is_live)
 
-    @staticmethod
-    async def ready() -> ServerReadyResponse:
+    async def ready(self) -> ServerReadyResponse:
         """Server ready endpoint.
 
         Returns:
             ServerReadyResponse: Server ready message.
         """
-        return ServerReadyResponse(ready=True)
+        is_ready = await self.dataplane.ready()
+        if not is_ready:
+            pass
```

Comment:
This should return 503 or 200 with ready=false?

Reply:
Now the server will return 503 if it is not ready